src/orderly/read.py

```diff
@@ -21,61 +9,100 @@ def orderly_read(path):
 # called once, and that is called at the top-level.
 #
 # The return value does nod at future extension though.
+def orderly_read(path):
+    src = path.read_text()
+    return _read_py(src)
+
+
 def _read_py(src):
-    ret = {"parameters": []}
-    for expr in src.body:
-        dat = _read_expr(expr)
-        if dat and isinstance(dat, ParametersCall):
-            if ret["parameters"]:
-                msg = f"Duplicate call to 'parameters()' on line {expr.lineno}"
+    module = ast.parse(src)
+    v = Visitor()
+    v.read_body(module.body)
+    return {"parameters": v.parameters or {}}
+
+
+class Visitor:
+    def __init__(self):
+        self.parameters = None
+
+    def read_body(self, stmts):
+        for stmt in stmts:
+            self._read_stmt(stmt)
+
+    def _read_stmt(self, stmt):
+        if (name := _match_orderly_call(stmt)) is not None:
```

Comment:
first time I've seen `:=`, though it's been in python since 3.8 apparently?

Reply:
Yeah it is quite recent. I like it because it makes it clear that `name` is only used inside the `if` block. Reminds me of the `if let Some(..) = ...` in Rust.

I'm happy to move the assignment to its own line above if you think this is confusing.